### PR TITLE
collect user mac hash for bi

### PR DESCRIPTION
**Pull request by yungez**
@ZhijunZhao, @aovechki

### CODE REVIEW

File: azure-documentdb-spring-boot-autoconfigure/src/main/java/com/microsoft/azure/autoconfigure/documentdb/DocumentDBAutoConfiguration.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import com.microsoft.azure.documentdb.ConnectionPolicy;
 import com.microsoft.azure.documentdb.ConsistencyLevel;
 import com.microsoft.azure.documentdb.DocumentClient;
+import com.microsoft.azure.spring.common.GetMacAddress;
```
Comment (ZhijunZhao) [inline azure-documentdb-spring-boot-autoconfigure/src/main/java/com/microsoft/azure/autoconfigure/documentdb/DocumentDBAutoConfiguration.java]: avoid verb in package name?

File: azure-documentdb-spring-boot-autoconfigure/src/test/java/com/microsoft/azure/autoconfigure/documentdb/PropertySettingUtil.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
 public class PropertySettingUtil {
     public static final String URI = "https://test.documents.azure.com:443/";
     public static final String KEY = "KeyString";
+    public static final boolean BIENABLED = true;
```
Comment (ZhijunZhao) [inline azure-documentdb-spring-boot-autoconfigure/src/test/java/com/microsoft/azure/autoconfigure/documentdb/PropertySettingUtil.java]: Now that true is by default, how about setting it to false?

File: azure-documentdb-spring-boot-starter/README.md [hunk 3]
```diff
@@ -97,6 +97,11 @@ public class SampleApplication implements CommandLineRunner {
 ```
 Autowired UserRepository interface, then can do save, delete and find operations.
 
+### BI data collection
+Microsoft would like to collect data about how users use this Spring boot starter. Data collected is like hash of mac address. Microsoft uses this information to improve our tooling experience. Participation is voluntary. If you don't want to participate, just simple disable it by setting below configuration in `application.properties`.
```
Comment (ZhijunZhao) [inline azure-documentdb-spring-boot-starter/README.md]: Data collected is like hash of mac address.
->
The only collected data is the hashed mac address. 

simple -> simply

File: azure-documentdb-spring-boot-autoconfigure/src/main/java/com/microsoft/azure/autoconfigure/documentdb/DocumentDBAutoConfiguration.java (hunk 1 above)
Comment (yungez) [inline azure-documentdb-spring-boot-autoconfigure/src/main/java/com/microsoft/azure/autoconfigure/documentdb/DocumentDBAutoConfiguration.java]: borrowed from Kevin's

File: azure-documentdb-spring-boot-autoconfigure/src/test/java/com/microsoft/azure/autoconfigure/documentdb/PropertySettingUtil.java (hunk 2 above)
Comment (yungez) [inline azure-documentdb-spring-boot-autoconfigure/src/test/java/com/microsoft/azure/autoconfigure/documentdb/PropertySettingUtil.java]: if true, will lose test coverage to useragent suffix setting.

**Pull request merged**